docs/index.rst

```diff
@@ -23,6 +23,41 @@ eventually converge to a stable API that maintains the common software engineeri
 will be marked as "alpha" indicating that their might be drastic changes over different releases.
 One should use alpha functionality with caution.
 
+How do we compare with NetworkX?
+--------------------------------
+We fashioned pywhy-graphs API based on NetworkX because NetworkX's API is stable, robust and
+has an existing community around it. Therefore, we expect all NetworkX users to have a relatively
+low learning curve when transitioning to pywhy-graphs. However, NetworkX does not currently support
+graphs with mixed-edges, so that is where the fundamental difference between the two APIs lie.
+
+In all the "NetworkX-like" functions related to edges, such as ``add_edge``, ``has_edge``,
+``number_of_edges``, etc. all have an additional keyword parameter, ``edge_type``. The edge
+type is specified by this parameter and internally, all pywhy-graphs graph classes are a
+composition of different networkx base graphs, :class:`networkx.DiGraph` and :class:`networkx.Graph`
+that map to a user-specified edge type. For example,
+
+.. code-block:: Python
+
+   import pywhy_graphs.networkx as pywhy_nx
+   import networkx as nx
+
+   # a mixed-edge graph is a composition of networkx graphs
+   # so we can create a representation of a graph with directed and
+   # bidirected edges using the MixedEdgeGraph class
+   G = pywhy_nx.MixedEdgeGraph()
+   G.add_edge_type(nx.DiGraph(), edge_type='directed')
+   G.add_edge_type(nx.Graph(), edge_type='bidirected')
+
+   # when we use networkx-like API, we usually will have to specify the edge type
+   G.add_edge(0, 1, edge_type='directed')
+
+Because of this feature, not all NetworkX algorithms will work with pywhy-graphs because
+they implicitly assume a single edge type. We implement common graph algorithms for
+mixed-edge graphs that have utility in causal inference in the :mod:`pywhy_graphs.algorithms`
+submodule. This is a similar design to NetworkX, where all graph classes have a relatively
+lightweight API designed solely for interfacing with nodes and edges, while complex traversal
```

Comment:
Are algorithms like finding connected components considered traversal algorithms?

Reply:
I guess so, but replaced the wording a bit here.